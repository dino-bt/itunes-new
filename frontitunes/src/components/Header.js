import React from "react";
import MobileHeader from "./mobileHeader"
import {useState} from "react"

//Here I use react router dom for the favourites page  to link favourites and home

const Header = () => {

  const[open, setOpen] = useState(false);


  return (
    <div>
    <nav className="nav">
      <div className="logo">ITUNES</div>
    
        <ul className = "nav-links">
          <li>
            <a id="homeheader" href="homeID">
              HOME
            </a>
          </li>

          <li>
            <a className="search" href="#searchId">
              SEARCH
            </a>
          </li>
          <li>
            <a className="favourites" href="#favouritesId">
              FAVOURITES
            </a>
          </li>
        </ul>
    <div className="burger" onClick= { () => setOpen(!open) }>
    <div className ="line1"></div>
    <div className="line2"></div>
    <div className="line3"></div>
    </div>
    
    </nav>
    
    {open === true ? ( 
    
    <MobileHeader   setOpen={setOpen}/>

    ) : (
        ""
      )}
      
      </div>
  );
};

export default Header;
