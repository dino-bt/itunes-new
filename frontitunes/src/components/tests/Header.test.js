import React from "react";
import renderer from "react-test-renderer";


import Header from "../Header";

it("tests the Header component", () => {
  const tree = renderer.create(<Header/>).toJSON();
  expect(tree).toMatchSnapshot();
});