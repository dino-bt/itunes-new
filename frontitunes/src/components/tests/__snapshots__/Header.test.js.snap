// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the Header component 1`] = `
<header>
  <div
    className="logo"
  >
    ITUNES
  </div>
  <div
    className="menu"
  >
    <ul>
      <li>
        <a
          href="homeID"
          id="homeheader"
        >
          HOME
        </a>
      </li>
      <li>
        <a
          className="search"
          href="#searchId"
        >
          SEARCH
        </a>
      </li>
      <li>
        <a
          className="favourites"
          href="#favouritesId"
        >
          FAVOURITES
        </a>
      </li>
    </ul>
  </div>
</header>
`;
